src/Signature/SignatureV4.php

```diff
@@ -76,6 +112,26 @@ public function signRequest(
         return $this->buildRequest($parsed);
     }
 
+    /**
+     * Get the headers that were used to pre-sign the request.
+     * Used for the X-Amz-SignedHeaders header.
+     *
+     * @param array $headers
+     * @return array
+     */
+    private function getPreSignHeaders(array $headers)
```

Comment:
Use `Presigned` as one word this is how it is used elsewhere (e.g. https://github.com/aws/aws-sdk-php/blob/master/src/Signature/SignatureV4.php#L189)